services/src/main/scala/cromwell/services/metadata/MetadataService.scala

```diff
@@ -145,6 +153,20 @@ object MetadataService {
     case WomPair(left, right) =>
       womValueToMetadataEvents(metadataKey.copy(key = metadataKey.key + ":left"), left) ++
         womValueToMetadataEvents(metadataKey.copy(key = metadataKey.key + ":right"), right)
+    case populated: WomMaybePopulatedFile =>
+      import mouse.all._
+      val secondaryFiles = populated.secondaryFiles.toEvents(metadataKey.copy(key = s"${metadataKey.key}:secondaryFiles"))
+
+      List(populated.valueOption |> toPrimitiveEvent(metadataKey, "value"),
+        populated.checksumOption |> toPrimitiveEvent(metadataKey, "checksum"),
+        populated.sizeOption |> toPrimitiveEvent(metadataKey, "size"),
+        populated.formatOption |> toPrimitiveEvent(metadataKey, "format"),
+        populated.contentsOption |> toPrimitiveEvent(metadataKey, "contents")
+      ) ++ secondaryFiles
```

Comment:
Assuming this is how we want to report Files / DIrectories in metadata, we might want to add a `class` field here too to make it clear what it is ? It would also help the `OutputManipulator` figure out what type of file it's dealing with

Reply:
I agree with Khalid

Reply:
`WomSingleFile`s serialize exactly the same so this shouldn't break any backwards compatibility as long as we haven't switched from `WomSingleFile`s to `WomMaybePopulatedFile`s where they really are `WomSingleFile`s ? ( @kshakir ?)

Reply:
At this second, afaik there are no `WomSingleFile` instance, under WDL, being converted to `WomMaybePopulatedFile` instances. Really hoping though that existing centaur tests will catch it if we do change this.

🤔 Although, now that I think about it, I don't believe we have any `metadata` file outputs in centaur, only `outputExpectations` that search-and-replace the `<<UUID>>` to get the correct dynamic path.

Reply:
Mokay, I was hoping too that something would break if it was not backward compatible but I'm less confident merging this if nothing tests it :/

Reply:
I can't find evidence that previous `WomSingleFile` from WDL workflows have been changed into `WomMaybePopulatedFile`s so I'm going to go ahead and merge this
I'll create a ticket though in the hope that we can get better testing of File output format in metadata in the future